src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/SystemUnicastIPAddressInformation.cs

```diff
@@ -0,0 +1,183 @@
+        private IPAddress _ipv4Mask;
+        private PrefixOrigin _prefixOrigin;
+        private SuffixOrigin _suffixOrigin;
+        private DuplicateAddressDetectionState _dadState;
+        private uint _validLifetime;
+        private uint _preferredLifetime;
+        private byte _prefixLength;
+
+        internal SystemUnicastIPAddressInformation(Interop.IpHlpApi.IpAdapterUnicastAddress adapterAddress)
+        {
+            IPAddress ipAddress = adapterAddress.address.MarshalIPAddress();
+            _innerInfo = new SystemIPAddressInformation(ipAddress, adapterAddress.flags);
+            _prefixOrigin = adapterAddress.prefixOrigin;
+            _suffixOrigin = adapterAddress.suffixOrigin;
+            _dadState = adapterAddress.dadState;
+            _validLifetime = adapterAddress.validLifetime;
+            _preferredLifetime = adapterAddress.preferredLifetime;
+            _dhcpLeaseLifetime = adapterAddress.leaseLifetime;
+
+            _prefixLength = adapterAddress.prefixLength;
+
+            // IPv6 returns 0.0.0.0 for consistency with down-level platforms.
+            if (ipAddress.AddressFamily == AddressFamily.InterNetwork)
+            {
+                _ipv4Mask = PrefixLengthToSubnetMask(_prefixLength, ipAddress.AddressFamily);
+            }
+        }
+
+        public override IPAddress Address { get { return _innerInfo.Address; } }
+
+        public override IPAddress IPv4Mask
+        {
+            get
+            {
+                // The IPv6 equivalent was never available on down-level platforms. 
+                // We've kept this behavior for legacy reasons. For IPv6 use PrefixLength instead.
+                if (Address.AddressFamily != AddressFamily.InterNetwork)
+                {
+                    return IPAddress.Any;
+                }
+
+                return _ipv4Mask;
+            }
+        }
+
+        public override int PrefixLength
+        {
+            get
+            {
+                return _prefixLength;
+            }
+        }
+
+        // The address is a cluster address and shouldn't be used by most applications.
+        public override bool IsTransient
+        {
+            get
+            {
+                return (_innerInfo.IsTransient);
+            }
+        }
+
+        // This address can be used for DNS.
+        public override bool IsDnsEligible
+        {
+            get
+            {
+                return (_innerInfo.IsDnsEligible);
+            }
+        }
+
+        public override PrefixOrigin PrefixOrigin
+        {
+            get
+            {
+                return _prefixOrigin;
+            }
+        }
+
+        public override SuffixOrigin SuffixOrigin
+        {
+            get
+            {
+                return _suffixOrigin;
+            }
+        }
+
+        // IPv6 only.  Specifies the duplicate address detection state. Only supported
+        // for IPv6. If called on an IPv4 address, will throw a "not supported" exception.
```

Comment:
Is this comment accurate? It doesn't look like there's a special code path for IPv4 addresses, so I don't think it will throw.